src/main/java/com/exactpro/epfast/decoder/message/DecodeMessage.java

```diff
@@ -0,0 +1,82 @@
+package com.exactpro.epfast.decoder.message;
+
+import com.exactpro.epfast.decoder.OverflowException;
+import io.netty.buffer.ByteBuf;
+
+import java.util.ArrayList;
+
+public class DecodeMessage {
```

Comment:
This class should be reusable. I see the following lifecycle:

1. MessageDecoder is created and initialized with list of instructions.
2. `decode()` method is called. At this point `message` is created.
3. `isReady()` method is called.
4. if decoder is ready `getMessageAndReset()` is called.
5. 2-5 are repeated